SFML/SFML
# Only build Cocoa example when using Xcode

## Description

The Cocoa example program requires Xcode so I disabled this when using any other CMake generator on macOS.

## Tasks


## How to test this PR?

Build SFML with SFML_BUILD_EXAMPLES=ON on macOS using Makefiles. Observe the following error:
```
xcode-select: error: tool 'ibtool' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```
